extensions/kind/src/create-cluster.ts

```diff
@@ -67,6 +67,116 @@ function getTags(tags: Tags): Tags {
   return tags;
 }
 
+const expectedErrorsInStartup: string[] = [
+  'ECONNREFUSED',
+  'ETIMEDOUT',
+  'ENOTFOUND',
+  'connect timeout',
+  'socket hang up',
+  '404',
+  '503 Service Unavailable',
+  'connection refused',
+];
+
+/**
+ * Handle errors during cluster startup. Only logs unexpected errors
+ */
+function handleStartupError(error: unknown, context: string): void {
+  const errorMessage = String(error);
+  const isExpectedError = expectedErrorsInStartup.some(expected =>
+    errorMessage.toLowerCase().includes(expected.toLowerCase()),
+  );
+
+  if (!isExpectedError) {
+    console.warn(`Unexpected error while ${context}: ${errorMessage}`);
+  }
+}
+
+/**
+ * Get Kubernetes client configuration
+ */
+function getKubeConfig(): KubeConfig {
+  const kc = new KubeConfig();
+  const kubeConfigPath = extensionApi.kubernetes.getKubeconfig().path;
+  kc.loadFromFile(kubeConfigPath);
+  return kc;
+}
+
+/**
+ * Wait for nodes to be ready using Kubernetes Watch API
+ */
+async function waitForNodesReady(): Promise<void> {
+  const kc = getKubeConfig();
+  const k8sApi = kc.makeApiClient(CoreV1Api);
+
+  let timeout = 0;
+  while (timeout < 300) {
+    try {
+      const nodes = await k8sApi.listNode();
+      if (nodes.items.length > 0) {
+        const allReady = nodes.items.every(node =>
+          node.status?.conditions?.some(condition => condition.type === 'Ready' && condition.status === 'True'),
+        );
+        if (allReady) return;
+      }
+    } catch (error) {
+      handleStartupError(error, 'waiting for nodes to be ready');
+    }
+
+    await new Promise(f => setTimeout(f, 100));
+    timeout++;
+  }
+
+  throw new Error('Nodes not ready within timeout');
+}
+
+/**
+ * Wait for system pods to be ready using Kubernetes client
+ */
+async function waitForSystemPodsReady(labelSelector: string): Promise<void> {
+  const kc = getKubeConfig();
+  const k8sApi = kc.makeApiClient(CoreV1Api);
+
+  let timeout = 0;
+  while (timeout < 300) {
+    try {
+      const pods = await k8sApi.listNamespacedPod({
+        namespace: 'kube-system',
+        labelSelector: labelSelector,
+      });
+
+      if (pods.items.length > 0) {
+        const allReady = pods.items.every(pod =>
+          pod.status?.conditions?.some(condition => condition.type === 'Ready' && condition.status === 'True'),
+        );
+        if (allReady) return;
+      }
+    } catch (error) {
+      handleStartupError(error, `waiting for system pods (${labelSelector})`);
+    }
+
+    await new Promise(f => setTimeout(f, 100));
+    timeout++;
+  }
+
+  throw new Error(`System pods with selector ${labelSelector} not ready within timeout`);
+}
+
+/**
+ * Wait for CoreDNS to be ready before installing ingress controller
+ * This prevents DNS timeout issues when Contour tries to resolve names
+ */
+export async function waitForCoreDNSReady(): Promise<void> {
```

Comment:
suggestion(blocking): following the suggestion in https://github.com/podman-desktop/podman-desktop/pull/14532/files#r2472458623

we should add the logger here


```suggestion
export async function waitForCoreDNSReady(logger?: extensionApi.Logger,): Promise<void> {
```

The log provided to the logger should be displayed in the terminal for the user; we should nicely logs all steps (wait for node, wait for pod ready...), errors etc.